src/execution/operator/persistent/physical_merge_into.cpp

```diff
@@ -505,9 +506,13 @@ SourceResultType PhysicalMergeInto::GetData(ExecutionContext &context, DataChunk
 
 		if (result != SourceResultType::FINISHED) {
 			return result;
-		}
-		if (chunk.size() != 0) {
-			return SourceResultType::HAVE_MORE_OUTPUT;
+		} else {
+			lstate.index++;
+			if (lstate.index < actions.size()) {
+				return SourceResultType::HAVE_MORE_OUTPUT;
+			} else {
+				return SourceResultType::FINISHED;
```

Comment:
I'm not sure if I follow why we need to increment the loop index here (instead of letting the `for` loop handle this). If the source operator returns `FINISHED`, can't we just `continue`?

And should we move the `DataChunk::Reset` calls to inside this loop?